phylo/src/optimisers/topo_optimiser.rs

```diff
@@ -0,0 +1,122 @@
+use log::{debug, info};
+
+use crate::likelihood::PhyloCostFunction;
+use crate::optimisers::{BranchOptimiser, PhyloOptimisationResult, PhyloOptimiser};
+use crate::phylo_info::PhyloInfo;
+use crate::tree::{NodeIdx, Tree};
+use crate::Result;
+
+pub struct TopologyOptimiser<'a, EM: PhyloCostFunction> {
+    pub(crate) epsilon: f64,
+    pub(crate) model: &'a EM,
+    pub(crate) info: PhyloInfo,
+}
+
+impl<'a, EM: PhyloCostFunction + Clone> PhyloOptimiser<'a, EM> for TopologyOptimiser<'a, EM> {
+    fn new(model: &'a EM, info: &PhyloInfo) -> Self {
```

Comment:
I recommend passing values instead of references here, and doing `.clone()` outside. This will make things cleaner

Reply:
This will be refactored in the next PR as per our discussion